packages/theme-check-common/src/liquid-doc/liquidDoc.ts

```diff
@@ -108,3 +108,27 @@ export function getSnippetDefinition(
     },
   };
 }
+
+function fixIndentation(text: string): string {
+  const lines = text.split('\n');
+
+  if (lines.length <= 1) return text;
+
+  const nonEmptyLines = lines.slice(1).filter((line) => line.trim().length > 0);
+  const indentLengths = nonEmptyLines.map((line) => {
+    const match = line.match(/^\s*/);
+    return match ? match[0].length : 0;
+  });
+
+  if (indentLengths.length === 0) return text;
+
+  const minIndent = Math.min(...indentLengths);
+
+  return [
+    lines[0],
+    ...lines.slice(1).map((line) => {
+      if (line.trim().length === 0) return line; // Skip empty lines
+      return line.slice(minIndent);
+    }),
+  ].join('\n');
+}
```

Comment:
There is a similar utility that exists in the prettier plugin but I chose to implement this here instead.

1) The only shared dependancy is `liquid-html-parser`
2) I explored moving it into a shared utility into `liquid-html-parser` and didn't feel that the benefit was > the cost of duplication here.
3) I highly doubt this will change, and if it does we don't want to change the prettier behaviour